### DISCUSSION TITLE
Hide the icon of a PlaceNode

### DISCUSSION THREAD

**Discussion by marcoma9023** [pelicanmapping/osgearth#2549] [Support Q&A] [answered]
As per the title,

after having created a custom widget similar to the PlaceNode, I would like to understand during the decluttering phase, i.e. when I have two overlapping PlaceNodes, how I can hide the icon behind it.

Currently the icon behind it is minimized, while the text is hidden.

I would like to have the same behavior for the icon too.

Ideas?

**Comment by gwaldron** [comment]
If you run the example `osgearth_tracks` you will see some options for controlling the decluttering behavior. These options are contained in the structure `ScreenSpaceLayoutOptions` and you can apply them by calling `ScreenSpaceLayout::setOptions`. 

(Please note that in the master branch, you will need to enable the CMake flag `OSGEARTH_BUILD_LEGACY_CONTROLS_API` for the `osgearth_tracks` example to build. We have not (yet) ported the demo to use ImGui.)

**Reply by marcoma9023** [reply, marked as answer]
Yes Glen, exactly what I was looking for.

```
osgEarth::Annotation::ScreenSpaceLayout::activate(_geodeStateSet.get());
osgEarth::ScreenSpaceLayout::setDeclutteringEnabled(true);
osgEarth::ScreenSpaceLayoutOptions options;
options.minAnimationAlpha() = 0.0;
options.minAnimationScale() = 0.0;
osgEarth::ScreenSpaceLayout::setOptions(options);
```

Thanks for the valuable reply.
